File: sol/3355/3355.py
class Solution:
    def isZeroArray(self, nums: List[int], queries: List[List[int]]) -> bool:
        n = len(nums)
        diff = [0] * (n + 1)
        for l, r in queries:
            diff[l] += 1
            diff[r + 1] -= 1
        cumsum = 0
        for i in range(n):
            cumsum += diff[i]
            if cumsum < nums[i]:
                return False
        return True

File: sol/3355/3355.cpp
struct Solution {
    bool isZeroArray(vector<int>& nums, vector<vector<int>>& queries) {
        int n = nums.size();
        vector<int> diff(n + 1, 0);
        for (auto &q : queries){
            ++diff[q[0]];
            --diff[q[1] + 1];
        }
        int cumsum = 0;
        for (int i = 0; i < n; ++i){
            cumsum += diff[i];
            if (cumsum < nums[i]){
                return false;
            }
        }
        return true;
    }
};
